src/System.Private.CoreLib/src/System/Reflection/Emit/DynamicMethod.cs

```diff
@@ -585,7 +580,17 @@ internal RTDynamicMethod(DynamicMethod owner, string name, MethodAttributes attr
             //
             public override string ToString()
             {
-                return ReturnType.FormatTypeName() + " " + FormatNameAndSig();
+                var sbName = new ValueStringBuilder(MethodNameBufferSize);
```

Comment:
Should we start with stack space instead of allocating?

Reply:
This change is fixing the easy to fix inefficiencies, but the methods are still pretty inefficient overall. 

I went back and forth on this, also considered to leave it on regular `StringBuilder`. I have settled on this as reasonable balance given what the rest looks like.